tensorflow/compiler/xla/stream_executor/rocm/rocm_driver.cc

```diff
@@ -1253,8 +1285,24 @@ static tsl::StatusOr<T> GetSimpleAttribute(hipDevice_t device,
     return false;
   }
 
-  *free_out = free;
-  *total_out = total;
+  uint64_t reserve = 0;
+  if (!GetReservedMemory(&reserve)){
+      LOG(ERROR) << "failed to reserved device memory for ROCm libraries";
+      return false;
+  }
+
+  VLOG(1) << "Device memory: " << total/1048576 << " MB total, "
+          << free/1048576 << " MB free, reserving " 
+          << reserve/1048576 << " MB";
+
+  // overflow check
+  if (free > std::numeric_limits<int64_t>::max()){
+      LOG(ERROR) << "free memory (" << free << ") is overflow int64_t";
```

Comment:
Reword. Maybe "Free memory (x) overflows int64_t".

Reply:
Thanks for the feedback! Do you want me to create another PR to revisit these words?